src/libraries/System.Security.Cryptography.Csp/src/System/Security/Cryptography/DSACryptoServiceProvider.Unix.cs

```diff
@@ -12,27 +12,41 @@ namespace System.Security.Cryptography
     public sealed class DSACryptoServiceProvider : DSA, ICspAsymmetricAlgorithm
     {
         private const int SHA1_HASHSIZE = 20;
+        private const int DefaultKeySize = 1024;
 
         private readonly DSA _impl;
         private bool _publicOnly;
 
-        private static KeySizes[] s_legalKeySizes =
+        private static KeySizes[] s_legalKeySizesWindowsCsp =
         {
             new KeySizes(512, 1024, 64)  // Use the same values as Csp Windows because the _impl has different values (512, 3072, 64)
         };
 
-        public DSACryptoServiceProvider() : base()
+        private static KeySizes[] s_legalKeySizesAndroid =
+        {
+            new KeySizes(1024, 1024, 0)  // Intersection of legal sizes on Android and Windows provider
+        };
+
+        public DSACryptoServiceProvider()
+            : this(DefaultKeySize)
         {
-            // This class wraps DSA
-            _impl = DSA.Create();
-            KeySize = 1024;
         }
 
         public DSACryptoServiceProvider(int dwKeySize) : base()
         {
             if (dwKeySize < 0)
                 throw new ArgumentOutOfRangeException(nameof(dwKeySize), SR.ArgumentOutOfRange_NeedNonNegNum);
 
+            // Depending on the platform, _impl's legal key sizes may be more restrictive than Windows provider.
+            //   DSAAndroid : (1024, 3072, 1024)
+            //   DSAOpenSsl : (512, 3072, 64)
+            //   DSASecurityTransforms : (512, 1024, 64)
+            //   Windows CSP : (512, 1024, 64)
+            // Use the most restrictive legal key sizes
+            LegalKeySizesValue = OperatingSystem.IsAndroid()
+                ? s_legalKeySizesAndroid
+                : s_legalKeySizesWindowsCsp;
```

Comment:
Save this computation as a static readonly so it only has to evaluate once?  (Sure, this type isn't popular, but may as well stay in good habits).